launchdarkly-android-client-sdk/src/main/java/com/launchdarkly/sdk/android/ConnectivityManager.java

```diff
@@ -76,19 +75,23 @@ class ConnectivityManager {
     // data is stored; 2. to implement additional logic that does not depend on what kind of data
     // source we're using, like "if there was an error, update the ConnectionInformation."
     private class DataSourceUpdateSinkImpl implements DataSourceUpdateSink {
+        private final ContextDataManager contextDataManager;
         private final AtomicReference<ConnectionMode> connectionMode = new AtomicReference<>(null);
         private final AtomicReference<LDFailure> lastFailure = new AtomicReference<>(null);
 
+        DataSourceUpdateSinkImpl(ContextDataManager contextDataManager) {
+            this.contextDataManager = contextDataManager;
+        }
+
         @Override
-        public void init(Map<String, DataModel.Flag> items) {
-            contextDataManager.initData(currentEvaluationContext.get(),
-                    EnvironmentData.usingExistingFlagsMap(items));
+        public void init(LDContext context, Map<String, DataModel.Flag> items) {
+            contextDataManager.initData(context, EnvironmentData.usingExistingFlagsMap(items));
```

Comment:
For reviewers:  The context was added to go along with the flag data.  This helps other parts of the system make decisions on whether or not to accept the flag data.  Ideally, this manager and the sources of this data could only exist when getting data from them makes sense, but a refactoring of the managers to be instead immutable instances and implement disposable is too risky at this time.